python/src/nanoarrow/_lib.pyx

```diff
@@ -903,3 +943,112 @@ cdef class ArrayStream:
     def allocate():
         base = ArrayStreamHolder()
         return ArrayStream(base, base._addr())
+
+
+cdef class DeviceArrayHolder:
+    """Memory holder for an ArrowDeviceArray
+
+    This class is responsible for the lifecycle of the ArrowDeviceArray
+    whose memory it is responsible. When this object is deleted,
+    a non-NULL release callback is invoked.
+    """
+    cdef ArrowDeviceArray c_array
+
+    def __cinit__(self):
+        self.c_array.array.release = NULL
+
+    def __dealloc__(self):
+        if self.c_array.array.release != NULL:
+          self.c_array.array.release(&self.c_array.array)
+
+    def _addr(self):
+        return <uintptr_t>&self.c_array
+
+cdef class DeviceHolder:
+    """Memory holder for an ArrowDevice
```

Comment:
Is this ArrowDevice concept something specific from nanoarrow? (I don't find it in https://arrow.apache.org/docs/dev/format/CDeviceDataInterface.html)

Reply:
Can you add some docstring then clarifying that? (to the Device class below)